src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -572,6 +595,8 @@
           (doseq [svc-id (keys services-by-id)] (swap! app-context assoc-in [:service-contexts svc-id] {}))
           (run-lifecycle-fns app-context s/init "init" ordered-services)
           (run-lifecycle-fns app-context s/start "start" ordered-services)
+          (when-let [restart-file (get-in-config (a/get-service this :ConfigService) [:global :restart-file])]
+            (inc-restart-counter! restart-file))
```

Comment:
I'm not 100% sure what my opinion is on this comment I'm about to make :)  but, just for consideration:

It might be a little more DRY if we moved the `when-let` part into the body of your function, and just passed in `this` as an extra argument to that function?